Purchases/Purchasing/StoreKit1/ProductsFetcherSK1.swift

```diff
@@ -167,24 +207,44 @@ private extension ProductsFetcherSK1 {
     // So we schedule a cancellation just in case, and skip it if all goes as expected.
     // More information: https://rev.cat/skproductsrequest-hangs
     func scheduleCancellationInCaseOfTimeout(for request: SKProductsRequest) {
-        queue.asyncAfter(deadline: .now() + .seconds(requestTimeoutInSeconds)) { [weak self] in
+        queue.asyncAfter(deadline: .now() + self.requestTimeout) { [weak self] in
             guard let self = self,
-                  let products = self.productsByRequests[request] else { return }
+                  let productRequest = self.productsByRequests[request] else { return }
 
             request.cancel()
 
-            Logger.appleError(Strings.storeKit.skproductsrequest_timed_out(after: self.requestTimeoutInSeconds))
-            guard let completionBlocks = self.completionHandlers[products] else {
+            Logger.appleError(Strings.storeKit.skproductsrequest_timed_out(
+                after: Int(self.requestTimeout.seconds.rounded())
+            ))
+            guard let completionBlocks = self.completionHandlers[productRequest.identifiers] else {
                 Logger.error("callback not found for failing request: \(request)")
                 return
             }
 
-            self.completionHandlers.removeValue(forKey: products)
+            self.completionHandlers.removeValue(forKey: productRequest.identifiers)
             self.productsByRequests.removeValue(forKey: request)
             for completion in completionBlocks {
-                completion(Set())
+                completion(.failure(ErrorCode.productRequestTimedOut))
             }
         }
     }
 
 }
+
+// swiftlint:disable identifier_name
+private extension DispatchTimeInterval {
+
+    var seconds: Double {
+        switch self {
+        case let .seconds(seconds): return Double(seconds)
+        case let .milliseconds(ms): return Double(ms) / 1000
+        case let .microseconds(ms): return Double(ms) / 1_000_000
+        case let .nanoseconds(ns): return Double(ns) / 1_000_000_000
+        case .never: return 0
+        @unknown default: fatalError("Unknown value: \(self)")
+        }
+    }
+
+}
```

Comment:
Feels like this one might be useful outside of the scope of this class, maybe we should move it to its own file? 
I could easily see us reusing the file to make it `Comparable`, for example. 

And perhaps we could add a description since `DispatchTimeInterval` is a very iOS-specific thing and might be confusing to devs used to other platforms? i.e.: 

```swift
/// `DispatchTimeInterval` can only be used by specifying a unit of time. 
/// This var allows us to easily convert any `DispatchTimeInterval` into seconds.
```

Reply:
Sounds good will do 👍